Tests/SentryTests/Networking/TestRequestManager.swift

```diff
@@ -19,7 +21,9 @@ public class TestRequestManager: NSObject, RequestManager {
     var responseDelay = 0.0
     public func add( _ request: URLRequest, completionHandler: SentryRequestOperationFinished? = nil) {
         
+        semaphore.wait()
         requests.append(request)
```

Comment:
L: is `requests.append(request)` an async call so you need a signal?

Reply:
No, it is not async. `add` can be called from multiple background threads so we need to sync the access to `requests`.